docs/howtos/model_surgery.rst

```diff
@@ -0,0 +1,232 @@
+Model Surgery
+==============================
+
+.. testsetup::
+
+  import functools
+  import numpy as np
+  import jax
+  from jax import lax, random, numpy as jnp
+  import flax
+  from flax import optim, traverse_util
+
+  from flax import linen as nn
+  from flax.core import unfreeze, freeze
+
+We will show how to get a flat dict of all the tensors, and then go back to a 
+nested, frozen dict. We show this for both Flax modules and optimizers.
```

Comment:
Suggestion:

- To avoid the "We ... show.... We show" repetition, maybe you can phrase it as follows:

```suggestion
We will show how to get a flat dict of all the tensors, and then go back to a 
nested, frozen dict. This will be demonstrated for both Flax modules and optimizers.
```